lib/lists.nix

```diff
@@ -687,6 +687,47 @@ rec {
    */
   unique = foldl' (acc: e: if elem e acc then acc else acc ++ [ e ]) [];
 
+  /* Check if list contains only unique elements.
+
+     Type: allUnique :: [a] -> bool
+
+     Example:
+       allUnique [ 3 2 3 4 ]
+       => false
+       allUnique [ 3 2 4 1 ]
+       => true
+   */
+  allUnique = lst: (builtins.length (unique lst) == builtins.length lst);
+
+  /* Check unique elements by sorting the list with a comparator and checking uniqueness with the previous element.
+     O(n) complexity on top of `sort`.
+
+     Type: sortUnique :: [a] -> [a]
+
+     Example:
+       sortUnique [ 3 2 4 1 ]
+       => false
+       sortUnique [ 3 2 3 4 ]
+       => true
+   */
+  sortUnique = lst:
+    let sortedList = lib.sort builtins.lessThan lst; in
+      (lib.foldl' (acc: e: {r = if e != acc.last then acc.r ++ [ e ] else acc.r; last = e;}) {r = []; last = (dummy: 0);} sortedList).r;
```

Comment:
@infinisil not sure if this is the perfect way... This is the first idea that came to my mind.

Reply:
Something like `findFirst id false (zipListsWith (fst: snd: fst != snd) sortedList (drop 1 sortedList))` might be nicer

Reply:
sorry, but I am unable to get this to work.

Reply:
Hmm you're right this doesn't work. And refactoring your implementation I'm noticing that there's a `++ [ e ]` in a loop, which actually makes this still an $\mathcal{O}(n^2)$ implementation. I think we should have https://github.com/NixOS/nixpkgs/pull/214021 first so this can be nicer. I'll push that PR a bit